src/httpserver.cpp

```diff
@@ -146,6 +148,10 @@ static GlobalMutex g_httppathhandlers_mutex;
 static std::vector<HTTPPathHandler> pathHandlers GUARDED_BY(g_httppathhandlers_mutex);
 //! Bound listening sockets
 static std::vector<evhttp_bound_socket *> boundSockets;
+//! Track active requests
+static Mutex g_requests_mutex;
```

Comment:
Maybe a `GlobalMutex` would be more appropriate here?

```suggestion
static GlobalMutex g_requests_mutex;
```

Reply:
I'm still newish to the repo, is there a convention on when we should use `Mutex` vs `GlobalMutex`?

Reply:
Depends on scope. For the global namespace the `GlobalMutex` is an appropriate option: https://github.com/bitcoin/bitcoin/blob/f3bc1a72825fe2b51f4bc20e004cef464f05b965/src/sync.h#L132-L141

Reply:
> Maybe a `GlobalMutex` would be more appropriate here?

This suggestion must fix the following [errors](https://api.cirrus-ci.com/v1/task/5176219491106816/logs/ci.log):
```
httpserver.cpp:221:13: error: calling function 'MaybeCheckNotHeld' requires negative capability '!g_requests_mutex' [-Werror,-Wthread-safety-analysis]
            LOCK(g_requests_mutex);
            ^
./sync.h:258:56: note: expanded from macro 'LOCK'
#define LOCK(cs) UniqueLock UNIQUE_NAME(criticalblock)(MaybeCheckNotHeld(cs), #cs, __FILE__, __LINE__)
                                                       ^
httpserver.cpp:218:9: error: calling function 'MaybeCheckNotHeld' requires negative capability '!g_requests_mutex' [-Werror,-Wthread-safety-analysis]
        LOCK(g_requests_mutex);
        ^
./sync.h:258:56: note: expanded from macro 'LOCK'
#define LOCK(cs) UniqueLock UNIQUE_NAME(criticalblock)(MaybeCheckNotHeld(cs), #cs, __FILE__, __LINE__)
                                                       ^
httpserver.cpp:481:9: error: calling function 'MaybeCheckNotHeld' requires negative capability '!g_requests_mutex' [-Werror,-Wthread-safety-analysis]
        WAIT_LOCK(g_requests_mutex, lock);
        ^
./sync.h:263:45: note: expanded from macro 'WAIT_LOCK'
#define WAIT_LOCK(cs, name) UniqueLock name(MaybeCheckNotHeld(cs), #cs, __FILE__, __LINE__)
                                            ^
3 errors generated.
```

Reply:
Done, errors are fixed. Thanks!